CHANGELOG.md

```diff
@@ -2,22 +2,56 @@
 
 See [RELEASE](./RELEASE.md) for workflow instructions.
 
-## UNRELEASED
+## v1.7.0
 
 ### Upgrade information
 
+Ensure you are running v1.6.x before upgrading to v1.7.0, if you skip v1.6.x your node will not be able to fully migrate
+its local state. Upgrading from v1.6.x to v1.7.0 will require a time consuming database migration that optimizes how
+the node stores at locally and will not allow you to downgrade to v1.6.x again.
+
 The command line flag `--scan-malfeasant-atxs` has been removed. All malfeasant ATXs before 1.6.0 have been marked as
 such and the node will continue to scan new ATXs for their validity.
 
 ### Highlights
 
+Database migrations now use a temporary copy of the existing database to speed up the process. This will require twice
+the local storage space for the duration of the migration.
```

Comment:
This is not necessarily true. Previously, large migrations were normally accompanied by `VACUUM`. Usual `VACUUM` actually does make a copy of the database under the hood, and it requires 2x extra space compared to the database size. The new mechanism replaces `VACUUM` with `VACUUM INTO` with a tmp version of the database, thus reducing the required extra space to just 1x.

With the new migration mechanism, when VACUUMing is not required (it's based on `database-vacuum-state` config value which is updated after adding large migrations), the old behavior is used, but with larger migrations, the amount of space required for temporary copies is actually reduced, at the same time improving the speed of migrations. Another important point is that in case of vacuuming, when the migrations are being applied to the temporary copy, they are made faster by disabling the journal/WAL and synchronous writers for that copy, as it can be thrown away in case of an incomplete migration.

Reply:
> This is not necessarily true. Previously, large migrations were normally accompanied by VACUUM. Usual VACUUM actually does make a copy of the database under the hood, and it requires 2x extra space compared to the database size. The new mechanism replaces VACUUM with VACUUM INTO with a tmp version of the database, thus reducing the required extra space to just 1x.

I'm not sure I understand. Before while migrating the DB might double in size, now instead we copy the whole DB into a new one with `VACUUM INTO` which will also require 2x the size (once for the original DB and once for the copy) or will it not?

> Another important point is that in case of vacuuming, when the migrations are being applied to the temporary copy, they are made faster by disabling the journal/WAL and synchronous writers for that copy, as it can be thrown away in case of an incomplete migration.

I already mentioned that the new approach is faster, I don't think we need to go into detail as to why in the changelog. 🙂

Reply:
Reformulated the paragraph to make it clearer what is needed for the migration and that this is only needed for larger migrations.